doc/source/ray-air/examples/pytorch_resnet_batch_prediction.ipynb

```diff
@@ -0,0 +1,398 @@
+    "s3.download_file(\n",
+    "    \"air-example-data-2\",\n",
+    "    \"imagenette2/imagenet_class_index.json\",\n",
+    "    \"/tmp/imagenet_class_index.json\",\n",
+    ")\n",
+    "\n",
+    "# Build mappings\n",
+    "idx_to_class = json.load(open(\"/tmp/imagenet_class_index.json\", \"r\"))\n",
+    "class_to_idx = {cls_name: int(index) for index, (cls_name, _) in idx_to_class.items()}\n",
+    "idx_to_class_name = {int(index): text for index, (_, text) in idx_to_class.items()}\n"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Next, let's define a preprocessor to crop and normalize images, as well as convert the class names to indices with the map that we just constructed. We'll use a {class}`BatchMapper <ray.data.preprocessors.BatchMapper>` to implement this custom preprocessing logic. `BatchMapper` takes in a user defined function and applies that as the data transformation to the input dataset, before feeding it into the model."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 7,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import numpy as np\n",
+    "import pandas as pd\n",
+    "from torchvision import transforms\n",
+    "from ray.data.preprocessors import BatchMapper\n",
+    "from typing import Dict\n",
+    "\n",
+    "\n",
+    "def preprocess(batch: Dict[str, np.ndarray]) -> Dict[str, np.ndarray]:\n",
+    "    transform = transforms.Compose(\n",
+    "        [\n",
+    "            transforms.ToPILImage(),\n",
+    "            transforms.Resize(256),\n",
+    "            transforms.CenterCrop(224),\n",
+    "            transforms.ToTensor(),\n",
+    "            transforms.Normalize(mean=[0.485, 0.456, 0.406], std=[0.229, 0.224, 0.225]),\n",
+    "        ]\n",
+    "    )\n",
+    "    batch[\"image\"] = np.array([transform(img).numpy() for img in batch[\"image\"]])\n",
+    "    batch[\"label\"] = np.array([class_to_idx[cls_name] for cls_name in batch[\"class\"]])\n",
+    "    return batch\n",
+    "\n",
+    "\n",
+    "preprocessor = BatchMapper(fn=preprocess, batch_format=\"numpy\")\n"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Build a BatchPredictor\n",
+    "\n",
+    "Now that we have our dataset loaded and preprocessed with [Ray Data](datasets), we're ready to construct our {class}`BatchPredictor <ray.data.batch_predictor.BatchPredictor>`! A {class}`BatchPredictor <ray.data.batch_predictor.BatchPredictor>` takes a checkpoint and a predictor class (e.g., {class}`~ray.train.torch.TorchPredictor`, {class}`~ray.train.tensorflow.TensorflowPredictor`) and provides an interface to run batch prediction on Ray {class}`~ray.data.Dataset`s. It will distribute the inference workload across multiple workers when calling `predict()` and run prediction on multiple shards of data in parallel. You can find more details in [Using Predictors for Inference](air-predictors).\n",
```

Comment:
The BatchPredictor links are not rendering correctly.

should be ray.train.batch_predictor.BatchPredictor